authentication/context.go

```diff
@@ -0,0 +1,28 @@
+package authentication
+
+import (
+	"context"
+	"errors"
+)
+
+type ctxKey int
+
+// CtxKey -
+const CtxKey ctxKey = 0
+
+// ErrNoCurrentUserInCtx -
+var ErrNoCurrentUserInCtx = errors.New("no current user in context")
+
+// GetCurrentUserFromCtx -
+func GetCurrentUserFromCtx(ctx context.Context) *CurrentUser {
```

Comment:
I would call this simply `GetFromContext`, so it becomes `authentication.GetFromContext(ctx)`.